utils/s3/client.cc

```diff
@@ -234,26 +235,84 @@ future<> map_s3_client_exception(std::exception_ptr ex) {
         auto e = std::current_exception();
         return make_exception_future<>(storage_io_error(EIO, format("S3 error ({})", e)));
     }
+}
+
+future<aws::aws_error> client::look_for_error(const http::reply& reply, input_stream<char>& in_, multipart_upload_completion is_mpu_completion) {
+    if (reply._status >= http::reply::status_type::multiple_choices || is_mpu_completion == multipart_upload_completion::yes) {
+        auto response_body = co_await util::read_entire_stream_contiguous(in_);
```

Comment:
Why do you need to read response body even if the reply status was "bad" from http perspective? Were this written in the patch comment, this would've been clear already

Reply:
> Many error responses contain additional structured data meant to be read and understood by a developer diagnosing programming errors. For example, if you send a Content-MD5 header with a REST PUT request that doesn't match the digest calculated on the server, you receive a BadDigest error. The error response also includes as detail elements the digest that the server calculated, and the digest that you told the server to expect. During development, you can use this information to diagnose the error. In production, a well-behaved program might include this information in its error log.

https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList

For example, there is 23 reasons to get the 403 response code and one definitely wants to know what exactly happened thats why I parse and log these error messages